pages/wiki/emulator.md

```diff
@@ -42,3 +42,41 @@ The previous command sets up a port forwarding so that you can connect to the em
 ```
 ssh -p 2222 ceres@localhost
 ```
+
+# Troubleshooting
+Here are some common problems and their solutions.
+
+## `qemu-system-i386: warning: host doesn't support requested feature: CPUID.80000001H:EDX.lm [bit 29]`
+This just means that you are running an i386 emulation on a 64-bit host processor.  For AsteroidOS, this will make no difference, but you can run `qemu-system-x86-64` instead of `qemu-system-i386` to bypass the error, which has to do with [long mode](https://en.wikipedia.org/wiki/Long_mode) on x86-64 machines.
+
+## `qemu-system-x86: -device virtio-vga,virgl=on: Property 'virtio-vga.virgl' not found`
+Newer versions of QEMU use `-device virtio-vga-gl` instead of `-device virtio-vga,virgl=on`, so if you get this message, use the newer form.
+
+## Video corruption using NVIDIA drivers
+There have been reported problems on machines using proprietary NVIDIA drivers.  One symptom is a black screen instead of the AsteroidOS screen which may or may not also show messages like these:
+
+```
+gl_version 46 - core profile enabled
+GLSL feature level 460
+failed to complete framebuffer 0x8cd7 asteroid-launch
+```
+Other symptoms include video artifacts (smearing, double screens) and crashing, usually with a message `Segmentation fault (core dumped)`.  If you're experiencing these and are using a proprietary NVIDIA driver, using `gtk` instead of `sdl` may work better.  An example command line:
```

Comment:
What about making `-display gtk,gl=on` the default instruction ? :)

Reply:
Not a bad idea, but `sdl` works better if it works at all.  With `gtk`, the video is not quite as sharp, at least on the two machines I have.

Reply:
Alright :)